libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java

```diff
@@ -0,0 +1,530 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.ai.qna;
+
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Strings;
+import com.microsoft.bot.ai.qna.models.RankerTypes;
+import com.microsoft.bot.ai.qna.models.Metadata;
+import com.microsoft.bot.ai.qna.models.QueryResult;
+import com.microsoft.bot.builder.IntentScore;
+import com.microsoft.bot.builder.Recognizer;
+import com.microsoft.bot.builder.RecognizerResult;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.Pair;
+import net.minidev.json.JSONObject;
+import okhttp3.OkHttpClient;
+
+/**
+ * IRecognizer implementation which uses QnAMaker KB to identify intents.
+ */
+// TODO: This class needs the missing StringExpression,
+// IntExpression, BoolExpression, ArrayExpression, ObjectExpression,
+// DialogContext classes
+public class QnAMakerRecognizer implements Recognizer {
```

Comment:
Here there is a disparity between Java/C#, the Recognizer is an iterface in Java but in C# is a [class](https://github.com/microsoft/botbuilder-dotnet/blob/014e3fe21a8d71e1cbaee039d20cb438f1e30148/libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs#L29). Here we are not implementing correctly the method of the interface

Reply:
I agree, I noticed that, but this should be changed by MS